Purchases/Purchasing/RCAttributionFetcher.m

```diff
@@ -163,23 +174,38 @@ - (void)postAttributionData:(NSDictionary *)data
         }
     }
 }
+- (void)postAppleSearchAdsAttributionCollectionIfNeeded {
+#if APP_TRACKING_TRANSPARENCY_AVAILABLE
+    if (@available(iOS 14, macos 11, tvos 14, *)) {
+        id<FakeATTrackingManager> trackingManagerClass = (id<FakeATTrackingManager>)NSClassFromString(@"ATTrackingManager");
+
+        if (trackingManagerClass) {
+            NSInteger authorizationStatus = [trackingManagerClass trackingAuthorizationStatus];
+            BOOL authorized = authorizationStatus == FakeATTrackingManagerAuthorizationStatusAuthorized;
+            if (!authorized) {
+                return;
+            }
+        }
+    }
+#endif
```

Comment:
this is the new bit, it just checks if the framework is available (at the OS and platform level), if it's included, and if authorization has been granted, and early returns if authorization could have been granted but wasn't. (i.e.: if the check was available, but the user hasn't granted permission because either they haven't been prompted, or they've been prompted and denied permission)